src/Bridge/Doctrine/Orm/Extension/PaginationExtension.php

```diff
@@ -143,7 +144,16 @@ public function supportsResult(string $resourceClass, string $operationName = nu
      */
     public function getResult(QueryBuilder $queryBuilder, string $resourceClass = null, string $operationName = null, array $context = [])
     {
-        $doctrineOrmPaginator = new DoctrineOrmPaginator($queryBuilder, $this->useFetchJoinCollection($queryBuilder, $resourceClass, $operationName));
+        $parts = $queryBuilder->getDQLPart('join');
+        $rootAlias = $queryBuilder->getRootAliases();
+        $query = $queryBuilder->getQuery();
+
+        // Only one alias, without joins, disable the DISTINCT on the COUNT
+        if (!isset($rootAlias[1]) && !($parts[$rootAlias[0] ?? ''] ?? true)) {
```

Comment:
`1 === count($queryBuilder->getAllAliases())` ?

Reply:
I tried to avoid count, what is better? :p

Reply:
Could it be possible that there are no root aliases? :P

So maybe

```php
if (1 >= \count($queryBuilder->getAllAliases())) {
```

Reply:
Why avoid count? :see_no_evil:

Reply:
perf :p. If we have more then 1 alias then we need distinct right?

Reply:
Sorry, I don't understand the perf concern. :laughing: At most we have a handful of aliases. It's `O(n)` but `n` is very small... (Actually, it seems like it might even be `O(1)` as PHP stores the number of elements.)